src/linker/Linker.Steps/LinkAttributesParser.cs

```diff
@@ -57,43 +57,83 @@ IEnumerable<CustomAttribute> ProcessAttributes (XPathNavigator nav, ICustomAttri
 					continue;
 
 				CustomAttribute customAttribute = CreateCustomAttribute (iterator, attributeType);
-				if (customAttribute != null)
+				if (customAttribute != null) {
+					_context.LogMessage ($"Assigning external custom attribute '{FormatCustomAttribute (customAttribute)}' instance to '{provider}'");
 					attributes.Add (customAttribute);
+				}
 			}
 
 			return attributes;
+
+			static string FormatCustomAttribute (CustomAttribute ca)
+			{
+				StringBuilder sb = new StringBuilder ();
+				sb.Append (ca.Constructor.GetDisplayName ());
+				sb.Append (" { args: ");
+				for (int i = 0; i < ca.ConstructorArguments.Count; ++i) {
+					var caa = ca.ConstructorArguments[i];
+					sb.Append ($"{caa.Type.GetDisplayName ()} {caa.Value}");
```

Comment:
You need to append the comma before the display name or the format will be wrong

Reply:
It's an arbitrary format for log, see the comma handling after type+constant pair

Reply:
Yes
What Im saying is that in i=0 you append the type name+value
In i=1 you append type name+value and then the coma
leaving you with "type name+valuetype name+value, "
the first strings will be together but there will be an extra comma at the end

Reply:
makes sense now